src/app/core/services/authentication-service/authentication.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { Router } from "@angular/router";
+
+import * as firebase from 'firebase/app';
+import { AngularFireAuth } from '@angular/fire/auth';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class AuthenticationService {
+
+  constructor(
+      private afAuth: AngularFireAuth,
+      private router: Router
+  ) { 
+  }
+
+  googleSignin(){
+    return new Promise<any>((resolve, reject) => {
+      let provider = new firebase.auth.GoogleAuthProvider();
+      provider.addScope('profile');
+      provider.addScope('email');
+      this.afAuth.auth
+      .signInWithPopup(provider)
+      .then(res => {
+        resolve(res);
+        this.router.navigate(['home']);
+      })
+    })
+  }
+
+  doRegister(value){
+    return new Promise<any>((resolve, reject) => {
```

Comment:
redundant promise creation. You can do the following `    return firebase.auth().createUserWithEmailAndPassword(value.email, value.password);`
It should work the same way. Though I would suggest not stick with promises but use Observables here. But it's totally up to you.

It's old [example ](https://stackoverflow.com/questions/38764578/rxjs-understanding-defer) but I think you can rewrite it into rxjs v6